src/main/java/uk/gov/hmcts/divorce/document/DocumentUtil.java

```diff
@@ -25,4 +31,39 @@ public static DivorceDocument divorceDocumentFrom(final DocumentInfo documentInf
             .documentType(documentType)
             .build();
     }
+
+    public static List<DivorceDocument> findDocumentsOfType(final CaseData caseData, final List<DocumentType> documentTypes) {
```

Comment:
Perhaps this method would be better if it took a `List<ListItem<DivorceDocument>>` rather than the CaseData and with a stream:

```
return documents == null ? emptyList() : documents.stream().filter(d -> isDocumentTypeOf(d, documentTypes).collect(...);
```

Reply:
I agree, I have also completely rewritten this method to add streaming and combine with conversion to letters method
https://github.com/hmcts/nfdiv-case-api/pull/429/commits/96867a2fc334ec70df3c9a16c38881bbabb5f75d